tsl/test/isolation/specs/addjob_with_dropped_check.spec

```diff
@@ -0,0 +1,40 @@
+# This file and its contents are licensed under the Timescale License.
+# Please see the included NOTICE for copyright information and
+# LICENSE-TIMESCALE for a copy of the license.
```

Comment:
Is there any particular reason you used an isolation test for this? Although this works, the isolation checks are intended to be used when you have multiple sessions that interact. As far as I can tell, this test can be written as a regression test and then be a lot easier to maintain.

Reply:
If you do it serially it will show something like "test_proc_with_check() does not exist"

`DROP FUNCTION` is interleaved with `SELECT add_job(`'s transaction in such a way that `add_job()` still gets the Oid of `test_proc_with_check` thanks to MVCC snapshot visiblity, but at the same time the function is removed before the actual execution of `add_job()`.

This was done to add coverage for the `function with OID %d does not exist` check in the C code.

Reply:
Unless I am mistaken, you should be able to use an arbitrary integer as OID in this case so you should be able to pick something that is completely out of range and use that as argument to the function.

```
mats=# create function foo(regproc) returns int as 'select $1::int' language sql;
CREATE FUNCTION
mats=# select foo(4711::regproc);
 foo  
------
 4711
(1 row)

mats=# select proname from pg_proc where oid = 4711;
 proname 
---------
(0 rows)
```

Reply:
Oh. Well that had not occurred to me at all. And you are absolutely right @mkindahl, this is enough to add test coverage without an isolation test. I had been drafting a response - not as succint as the one @mfundul provided, but something along the lines of why I think isolation tests are needed because of MVCC.